src/pe/header.rs

```diff
@@ -3,24 +3,60 @@ use crate::pe::{optional_header, section_table, symbol};
 use crate::strtab;
 use alloc::vec::Vec;
 use log::debug;
-use scroll::{IOread, IOwrite, Pread, Pwrite, SizeWith};
+use scroll::{ctx, IOread, IOwrite, Pread, Pwrite, SizeWith};
 
 /// DOS header present in all PE binaries
 #[repr(C)]
-#[derive(Debug, PartialEq, Copy, Clone, Default)]
+#[derive(Debug, PartialEq, Copy, Clone, Default, Pwrite)]
 pub struct DosHeader {
     /// Magic number: 5a4d
     pub signature: u16,
-    /// Pointer to PE header, always at offset 0x3c
+    /// e_cblp
+    pub bytes_on_last_page: u16,
+    /// e_cp
+    pub pages_in_file: u16,
+    /// e_crlc
+    pub relocations: u16,
+    /// e_cparhdr
+    pub size_of_header_in_paragraphs: u16,
+    /// e_minalloc
+    pub minimum_extra_paragraphs_needed: u16,
+    /// e_maxalloc
+    pub maximum_extra_paragraphs_needed: u16,
+    /// e_ss
+    pub initial_relative_ss: u16,
+    /// e_sp
+    pub initial_sp: u16,
+    /// e_csum
+    pub checksum: u16,
+    /// e_ip
+    pub initial_ip: u16,
+    /// e_cs
+    pub initial_relative_cs: u16,
+    /// e_lfarlc
+    pub file_address_of_relocation_table: u16,
+    /// e_ovno
+    pub overlay_number: u16,
+    /// e_res[4]
+    pub reserved: [u16; 4],
+    /// e_oemid
+    pub oem_id: u16,
+    /// e_oeminfo
+    pub oem_info: u16,
+    /// e_res2[10]
+    pub reserved2: [u16; 10],
+    /// e_lfanew: pointer to PE header, always at offset 0x3c
     pub pe_pointer: u32,
```

Comment:
not super familar with DOS Header or PE, but how is it possible there are extra fields in between pe_pointer now?

Reply:
Because you were jumping directly after the pe_pointer IIRC?

Reply:
ah yes, i see now:
```rust
        let pe_pointer = bytes
            .pread_with(PE_POINTER_OFFSET as usize, scroll::LE)
            .map_err(|_| {
                error::Error::Malformed(format!(
                    "cannot parse PE header pointer (offset {:#x})",
                    PE_POINTER_OFFSET
                ))
            })?;

```